scripts/dumps/bulk_genome_dump_from_core.py

```diff
@@ -0,0 +1,337 @@
+                "bsub",
+                "-W",
+                "1:00",
+                "-R",
+                "rusage[mem=4096]",
+                "-J",
+                job_name,
+                f"-o {stdout_file}",
+                f"-e {stderr_file}",
+            ] + command
+
+    logging.info("Running: %s", " ".join(command))
+    with subprocess.Popen(
+        command,
+        shell=shell,
+        cwd=work_dir,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    ) as process:
+        output, stderr = process.communicate()
+
+        if process.returncode != 0:
+            out = f"stdout={output}, stderr={stderr}"
+            logging.error(
+                "Command %s exited %d: %s", " ".join(command), process.returncode, out
+            )
+            raise RuntimeError(
+                f"Command {' '.join(command)} exited {process.returncode}: {out}"
+            )
+
+        logging.info("Successfully ran: %s", " ".join(command))
+        return output.strip()
+
+
+def download_file(
+    host,
+    port,
+    core_db,
+    fasta_filename,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    genome_component="",
+    mask="soft",
+):
+    """
+    Download the FASTA file from the core DB using a PERL script `dump_genome_from_core.pl`.
+
+    Args:
+        host (str): The DB host name.
+        port (str): The port number.
+        core_db (str): The core_db name.
+        fasta_filename (str): The name given for the FASTA file.
+        mask (str): The mask format for the FASTA file.
+
+    Returns:
+        str: The output of the subprocess call to download the FASTA file.
+    """
+    try:
+        ensembl_root_dir = os.environ["ENSEMBL_ROOT_DIR"]
+    except KeyError:
+        logging.exception("Environment variable ENSEMBL_ROOT_DIR not set")
+        raise
+
+    script_dir = os.path.join(
+        ensembl_root_dir,
+        "ensembl-compara",
+        "scripts",
+        "dumps",
+    )
+    script = "dump_genome_from_core.pl"
+
+    try:
+        perl_call = [
+            "perl",
+            os.path.join(script_dir, script),
+            "--core_db",
+            core_db,
+            "--host",
+            host,
+            "--port",
+            str(port),
+            "--mask",
+            mask,
+            "-user",
+            "ensro",
+            "--outfile",
+            fasta_filename,
+        ]
+
+        # Conditionally add the genome component argument
+        if genome_component:
+            perl_call += ["--genome-component", genome_component]
+
+        logging.info("perl_call=%s", perl_call)
+        return subprocess_call(
+            command=perl_call,
+            use_job_scheduler=True,
+            job_name=f"{core_db}_{genome_component}",
+            stderr_file=stderr_file,
+            stdout_file=stdout_file,
+        )
+
+    except Exception as e:
+        logging.error("An unexpected error occurred: %s", e)
```

Comment:
Could `logging.exception` be used here? 
```suggestion
        logging.exception("An unexpected error occurred: ")
```